tests/src/FunctionalJavascript/ContentProjectRenderTest.php

```diff
@@ -0,0 +1,441 @@
+
+    // Select the content column next to the in-page navigation.
+    $project_content = $assert_session->elementExists('css', '.col-md-9');
+
+    $this->assertProjectStatusTimestampsAsDateStrings('2020-05-10 00:00:00', '2025-05-16 00:00:00');
+    $this->assertProjectStatusVisible();
+
+    $contributions_chart = $assert_session->elementExists('css', '.bcl-project-contributions .circular-progress');
+    // The correct value would be 35, but it is rounded up to 40, because no
+    // utility classes are available for smaller increments.
+    $this->assertSame('40', $contributions_chart->getAttribute('data-percentage'));
+
+    // Select the description blocks inside the Project details.
+    $description_lists = $project_content->findAll('css', '.grid-3-9');
+    $this->assertCount(4, $description_lists);
+
+    $description_list_assert = new DescriptionListAssert();
+
+    // Assert budget list group.
+    $description_list_assert->assertPattern([
+      'items' => [
+        [
+          'term' => 'Overall budget',
+          'definition' => '€200,00',
+        ],
+        [
+          'term' => 'EU contribution',
+          'definition' => '€70,00',
+        ],
+      ],
+    ], $description_lists[0]->getHtml());
+
+    // Assert details list group.
+    $description_list_assert->assertPattern([
+      'items' => [
+        [
+          'term' => 'Website',
+          'definition' => 'Example website',
+        ],
+        [
+          'term' => 'Funding programme',
+          'definition' => 'Anti Fraud Information System (AFIS)',
+        ],
+        [
+          'term' => 'Reference',
+          'definition' => 'Project reference',
+        ],
+      ],
+    ], $description_lists[1]->getHtml());
+
+    // Assert coordinators list group.
+    $description_list_assert->assertPattern([
+      'items' => [
+        [
+          'term' => 'Coordinators',
+          'definition' => 'coordinator',
+        ],
+      ],
+    ], $description_lists[2]->getHtml());
+
+    // Assert participants list group.
+    $description_list_assert->assertPattern([
+      'items' => [
+        [
+          'term' => 'Name',
+          'definition' => 'participant',
+        ],
+        [
+          'term' => 'Address',
+          'definition' => 'Belgium',
+        ],
+        [
+          'term' => 'Contribution to the budget',
+          'definition' => '€22,30',
+        ],
+      ],
+    ], $description_lists[3]->getHtml());
+
+    // Set a project period that is fully in the past.
+    $this->setProjectDateRange($node, '2019-03-07', '2019-03-21');
+    $node->save();
+    // Manipulate the server time to a date even further in the past.
+    $this->getTestTimeObject()->setTime('2017-01-01');
+    $this->drupalGet($node->toUrl());
+
+    $this->assertProjectStatusTimestampsAsDateStrings('2019-03-07 00:00:00', '2019-03-22 00:00:00');
+    $this->assertProjectStatusVisible();
+    $this->assertProjectStatus('bg-dark', 'Closed');
```

Comment:
Selenium uses the current date, the server uses the manipulated date.
The tests would fail without javascript (and they do), but they pass with javascript.
That's the point of the ->setTime().
I should add a comment.

I also wanted to add assertions for non-js situation, but did not find how to access the unmodified html.